Neos.Utility.Arrays/Classes/ValueAccessor.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Neos\Utility;
+
+/*
+ * This file is part of the Neos.Utility.Arrays package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Type safe accessing of values from nested arrays
+ */
+class ValueAccessor
+{
+    public function __construct(
+        public readonly mixed   $value,
+        public readonly ?string $pathinfo = null
+    ) {
+    }
```

Comment:
i would like to propose a follow up, where we make this constructor internal and expose those two static factories:

```
// @internal
public static function fromValueWithPathInfo

// @api
public static function fromValue
```